src/usb.c

```diff
@@ -0,0 +1,1619 @@
+                                             XDG_DESKTOP_PORTAL_ERROR_NOT_ALLOWED,
+                                             "Not allowed to create USB sessions");
+      return G_DBUS_METHOD_INVOCATION_HANDLED;
+    }
+
+  impl_request = xdp_dbus_impl_request_proxy_new_sync (g_dbus_proxy_get_connection (G_DBUS_PROXY (usb_impl)),
+                                                       G_DBUS_PROXY_FLAGS_NONE,
+                                                       g_dbus_proxy_get_name (G_DBUS_PROXY (usb_impl)),
+                                                       request->id,
+                                                       NULL,
+                                                       &error);
+  if (!impl_request)
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return G_DBUS_METHOD_INVOCATION_HANDLED;
+    }
+
+  g_variant_builder_init (&options_builder, G_VARIANT_TYPE_VARDICT);
+  if (!xdp_filter_options (arg_options,
+                           &options_builder,
+                           usb_access_devices_options,
+                           G_N_ELEMENTS (usb_access_devices_options),
+                           &error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return G_DBUS_METHOD_INVOCATION_HANDLED;
+    }
+  options = g_variant_builder_end (&options_builder);
+
+  sender_info = usb_sender_info_from_request (self, request);
+  g_assert (sender_info != NULL);
+
+  if (sender_info->sender_state != USB_SENDER_STATE_DEFAULT)
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             XDG_DESKTOP_PORTAL_ERROR,
+                                             XDG_DESKTOP_PORTAL_ERROR_FAILED,
+                                             "Cannot call AcquireDevices() with an unfinished "
+                                             "call to AcquireDevices()");
+      return G_DBUS_METHOD_INVOCATION_HANDLED;
+    }
+
+  /* Validate devices */
+  if (!filter_access_devices (self, sender_info, arg_devices, &filtered_devices, &error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      return G_DBUS_METHOD_INVOCATION_HANDLED;
+    }
+
+  request_set_impl_request (request, impl_request);
+  request_export (request, g_dbus_method_invocation_get_connection (invocation));
+
+  sender_info->sender_state = USB_SENDER_STATE_ACQUIRING_DEVICES;
+
+  xdp_dbus_impl_usb_call_acquire_devices (usb_impl,
+                                          request->id,
+                                          arg_parent_window,
+                                          xdp_app_info_get_id (request->app_info),
+                                          g_steal_pointer (&filtered_devices),
+                                          g_steal_pointer (&options),
+                                          NULL,
+                                          usb_access_devices_cb,
```

Comment:
```suggestion
                                          usb_devices_acquire_cb,
```
or `on_usb_devices_acquired`